hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java

```diff
@@ -16,39 +15,34 @@
 import org.slf4j.LoggerFactory;
 
 public class MongoBackendResolver extends AbstractBackendResolver {
+
   private static final Logger LOGGER = LoggerFactory.getLogger(MongoBackendResolver.class);
+
   private static final String RAW_MONGO_NAMESPACE = "NAMESPACE";
-  private static final String MONGO_ADDRESS_ATTR = RawSpanConstants.getValue(Mongo.MONGO_ADDRESS);
-  private static final String MONGO_URL_ATTR = RawSpanConstants.getValue(Mongo.MONGO_URL);
 
   public MongoBackendResolver(FQNResolver fqnResolver) {
     super(fqnResolver);
   }
 
   @Override
   public Optional<Entity> resolveEntity(Event event, StructuredTraceGraph structuredTraceGraph) {
-    if (SpanAttributeUtils.containsAttributeKey(event, MONGO_ADDRESS_ATTR) ||
-        SpanAttributeUtils.containsAttributeKey(event, MONGO_URL_ATTR)) {
-      String backendURI = null;
-      // check if this is from uninstrumeted agent first
-      if (SpanAttributeUtils.containsAttributeKey(event, MONGO_ADDRESS_ATTR)) {
-        backendURI = SpanAttributeUtils.getStringAttribute(event, MONGO_ADDRESS_ATTR);
-      } else if (SpanAttributeUtils.containsAttributeKey(event, MONGO_URL_ATTR)) {
-        backendURI = SpanAttributeUtils.getStringAttribute(event, MONGO_URL_ATTR);
-      }
-
-      if (backendURI == null) {
-        LOGGER.debug("Detected MONGO span event, but unable to extract the URI. Span Event: {}",
-            event);
-        return Optional.empty();
-      }
-
-      final Builder entityBuilder = getBackendEntityBuilder(BackendType.MONGO, backendURI, event);
-      setAttributeIfExist(event, entityBuilder, RAW_MONGO_NAMESPACE);
-      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_NAMESPACE));
-      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_OPERATION));
-      return Optional.of(entityBuilder.build());
+    Optional<String> backendURI = DbTagResolver.getMongoURI(event);
+
+    if (backendURI.isEmpty()) {
+      return Optional.empty();
+    }
+
+    if (StringUtils.isEmpty(backendURI.get())) {
```

Comment:
should be this if merged with the one above?

Reply:
yes, will fix